llvm/utils/repo/repo.cmake

```diff
@@ -80,8 +80,8 @@ set (CMAKE_LINKER ${utils_dir}/link.py)
 set (CMAKE_C_LINKER ${CMAKE_LINKER})
 set (CMAKE_CXX_LINKER ${CMAKE_LINKER})
 
-set (CMAKE_C_LINK_EXECUTABLE "${utils_dir}/link.py <FLAGS> <CMAKE_C_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> -o <TARGET> <LINK_LIBRARIES>")
-set (CMAKE_CXX_LINK_EXECUTABLE "${utils_dir}/link.py <FLAGS> <CMAKE_CXX_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> -o <TARGET> <LINK_LIBRARIES>")
+set (CMAKE_C_LINK_EXECUTABLE "${utils_dir}/link.py <FLAGS> <CMAKE_C_LINK_FLAGS> <OBJECTS> <LINK_LIBRARIES> <LINK_FLAGS> -o <TARGET>")
+set (CMAKE_CXX_LINK_EXECUTABLE "${utils_dir}/link.py <FLAGS> <CMAKE_CXX_LINK_FLAGS> <OBJECTS> <LINK_LIBRARIES> <LINK_FLAGS> -o <TARGET>")
```

Comment:
Why has the LINK_FLAGS variable moved after LINK_LIBRARIES?

Reply:
The same reason as the LINK_FLAGS variable moved after OBJECTS.

The `-lc_elf` is in the end of LINK_FLAGS.  LINK_LIBRARIES include some object files, which use some functions defined in the libc_elf.a or libc++ libraries. As mentioned in [Link order of libraries](https://www.linuxtopia.org/online_books/an_introduction_to_gcc/gccintro_18.html), the -lc++, -lc++abi, -lunwind and -lc_elf need to be put after LINK_LIBRARIES to avoid undefiend functions errors.

An example is shown below. The link command shows LINK_LIBRARIES are moved after LINK_FLAGS.
```
~/github/llvm-project-prepo/llvm/utils/repo/link.py -stdlib=libc++ -isystem /include/c++/v1 -D__MUSL__ -nostdinc -nostdinc++ --sysroot /home/maggie/musl -isystem /home/maggie/musl/include -I/home/maggie/LLVM/include/c++/v1/ -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wno-noexcept-type -Wnon-virtual-dtor -Wno-comment -fdiagnostics-color  -O3  tools/pstore/unittests/adt/CMakeFiles/pstore-adt-unit-tests.dir/test_chunked_vector.cpp.o tools/pstore/unittests/adt/CMakeFiles/pstore-adt-unit-tests.dir/test_error_or.cpp.o tools/pstore/unittests/adt/CMakeFiles/pstore-adt-unit-tests.dir/test_small_vector.cpp.o tools/pstore/unittests/adt/CMakeFiles/pstore-adt-unit-tests.dir/test_sstring_view.cpp.o  lib/libLLVMSupport.a  lib/libgtest_main.a  lib/libgtest.a  lib/libpstore-adt.a  lib/libLLVMSupport.a  -lrt  -ldl  -lm  lib/libLLVMDemangle.a  lib/libpstore-support.a -nostdlib -nodefaultlibs -static --sysroot /home/maggie/musl -L /home/maggie/musl/lib -stdlib=libc++ -L /home/maggie/LLVM/lib /home/maggie/LLVM/lib/linux/clang_rt.crtbegin-x86_64.o.elf /home/maggie/LLVM/lib/linux/clang_rt.crtend-x86_64.o.elf -lc++ -lc++abi -lunwind -L /home/maggie/LLVM/lib/linux -lclang_rt.builtins-x86_64 /home/maggie/musl/lib/crt1.t.o /home/maggie/musl/lib/crt1_asm.t.o -lc_elf -Wl,-allow-shlib-undefined    -Wl,-O3 -Wl,--gc-sections -o tools/pstore/unittests/adt/pstore-adt-unit-tests
```
We can see the LINK_LIBRARIES (lib/libLLVMSupport.a  lib/libgtest_main.a  lib/libgtest.a  lib/libpstore-adt.a  lib/libLLVMSupport.a  -lrt  -ldl  -lm  lib/libLLVMDemangle.a  lib/libpstore-support.a) are moved before the LINK_FLAGS. It resolved a lot of missing undefined function errors.

However, I got another link error: missing main function definition. The reason is the musl crts functions need to be linked before the lib/libgtest_main.a. The repo.cmake is modified in the commit [c55d0ef14](https://github.com/SNSystems/llvm-project-prepo/pull/149/commits/c55d0ef14a358c507a4aec7f8b66fde61aad268c) to solve the issue.

With the latest repo.cmake, the pstore-write passed the build stage but there are two unit tests failed.